docs/zkasm/infra.md

```diff
@@ -0,0 +1,64 @@
+# General Overview
+
+Firstly, we aim to implement the following types of tests and benchmarks:
+- Wasm spectest: Programs generated from Wasm spectest.
+- Benchmarks: Programs we manually write in Wasm or Rust and compile to Wasm.
+
+Both types of programs will undergo the same pipeline:
```

Comment:
It's true that so far, these two types of programs have been handled similarly, but going forward, this is likely to change to give a better experience with both of these tools.
For example, tests must be easy and fast to run and must present aggregated reports across many tests. Benchmarks, on the other hand, can take a long time to run, will be compiled with different optimization flags, and will likely become parameterized by the size of the input (e.g. N-th Fibonacci number with N = 1..10^6).

So in this document, I would focus specifically on our goals for the test suite and leave benchmarks out of scope.